src/MeshTransfer.f90

```diff
@@ -6,7 +6,54 @@
 ! of Michigan and Oak Ridge National Laboratory.  The copyright and license    !
 ! can be found in LICENSE.txt in the head directory of this repository.        !
 !++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
-!> @brief A Fortran 2003 module implementing some basic sorting algorithms.
+!> @brief A Fortran 2003 module implementing mesh transfer methods.
+!>
+!>  This package defines a base mesh transfer algorithm and a cartesian and
+!>  cylindrical transfer in 1D.  There are multiple options available for mesh
+!>  transfer including: point-to-point, point-to-volume, point-to-continuous,
+!>  volume-to-point, volume-to-volume, volume-to-continuous, continuous-to-point
+!>  continuous-to-volume, and continuous-to-continuous.
+!>
+!>  The point-to-point transfer can be done using linear interpolation on the
+!>  incoming mesh or through a projection to an user-defined order polynomial.
+!>  For the linear interpolation method, data points outside the input data range
+!>  will return the boundary value.
+!>
+!>  The point-to-volume transfer can be done by integrating the linear function
+!>  between the points or by integrating an user-defined order polynomial.
+!>
+!>  The point-to-continuous transfer projects the points to a defined order
+!>  polynomial.
+!>
+!>  The volume-to-point transfer returns the values for the volume in which the
+!>  points fall into or can use a user-defined order polynomial which minimizes
+!>  the error that preserves the volume average quantities.
+!>
+!>  The volume-to-volume transfer returns the integral values for flat
+!>  volume-averaged input data or a user-defined order polynomial fit which
+!>  minimizes the error in the volume average input data.
+!>
+!>  The volume-to-continuous transfer returns the defined polynomial fit which
+!>  minimizes the error in the volume average input data.
+!>
+!>  The continuous-to-point transfer returns point values on the continuous
+!>  input polynomial.
+!>
+!>  The continuous-to-volume transfer returns volume average values from the
+!>  continuous input polynomial.
+!>
+!>  The continuous-to-continuous will return updated coefficients based on the
+!>  outgoing moments requested.
+!>
+!>  In all cases, the user specifies the input mesh or moments and output mesh
+!>  or moments and to use the native implementations or a polynomial projection
+!>  transfer during initialization.  A mesh transfer matrix or linear system
+!>  will be used to transfer input data onto the output mesh.  The call to
+!>  transfer will transform the input data onto the output mesh using the transfer
+!>  matrix and/or linear system which was defined during init.  As long as the
+!>  incoming and outgoing mesh/moments are the same, transfer can be called
+!>  an arbitrary number of times very efficiently.  If the mesh is different,
+!>  a new transfer object will be needed.
```

Comment:
It'd be cool to insert latex-only blocks for the actual mathematical formulations. See XenonSamarium.f90 for an example.

This is something I'd like to see us do more of, because it gives us the mathematical "requirements" in the comments.

Reply:
I'm not entirely sure how to represent this mathematically outside of saying ```out=MT*in``` or ```LS%A*out = in```.  Do you have something specific that you think would add value to the description?